lib/stat_tracker.rb

```diff
@@ -93,21 +97,61 @@ def average_goals_per_game
   end
   
   def average_goals_by_season
-    hash = {}
+    all_goals = Hash.new {|h, k| h[k] = [] }
     games.each do |game|
-      if hash[game.season]
-        hash[game.season] <<  sum_of_goals(game)
-      else
-        hash[game.season] = [sum_of_goals(game)]
-      end
+      all_goals[game.season] <<  sum_of_goals(game)
     end
-    hash.each do |key, value|
+
+    all_goals.each do |key, value|
       average = (value.sum / value.size.to_f).round(2)
-      hash[key] = average
+      all_goals[key] = average
     end
   end
     
-  def count_of_teams
-    @teams.count
+
+  def highest_scoring_visitor
+    tally_scores = Hash.new {|h, k| h[k] = [] }
+    games.each do |game|
+      tally_scores[game.away_team_id] <<  game.away_goals.to_i
+    end
+    average_and_lookup(tally_scores, :max)
+  end
+
+  def highest_scoring_home_team
+    tally_scores = Hash.new {|h, k| h[k] = [] }
+    games.each do |game|
+      tally_scores[game.home_team_id] <<  game.home_goals.to_i
+    end
+    average_and_lookup(tally_scores, :max)
+  end
+
+  def lowest_scoring_visitor
+    tally_scores = Hash.new {|h, k| h[k] = [] }
+    games.each do |game|
+      tally_scores[game.away_team_id] <<  game.away_goals.to_i
+    end
+    average_and_lookup(tally_scores, :min)
+  end
+  
+  def lowest_scoring_home_team 
+    tally_scores = Hash.new {|h, k| h[k] = [] }
+    games.each do |game|
+      tally_scores[game.home_team_id] <<  game.home_goals.to_i
+    end
+    average_and_lookup(tally_scores, :min)
+  end
+
+  ### HELPER FUNCTION ###
```

Comment:
I like the helper method! It made me realize how we could refactor our best and worst offense methods, which is good to see.